lib/src/list/built_list.dart

```diff
@@ -72,7 +72,7 @@ class BuiltList<E> implements Iterable<E>, BuiltIterable<E> {
   @override
   bool operator ==(dynamic other) {
     if (identical(other, this)) return true;
-    if (other is! BuiltList) return false;
+    if (other is! _BuiltList) return false;
```

Comment:
This should stay `BuiltList`, I think. operator== only depends on public parts of the API.

Reply:
Looks good, thanks!

I think operator== should just check for the public type, since it'll work with any implementation.

Changing the other classes too sounds good.

Just a heads-up, this is going to break `built_value` serialization, because that depends on `runtimeType`. I just tried it though and the fix should be straightforward, and fortunately also forwards/backwards-compatible. It'll mean we need a new release of `built_value` before releasing this change--I should be able to do that.

Thanks!

Reply:
Regarding built_value: Will my 3rd party class work with built_value_generator if I implement `BuiltList`, or do I have to implement `Built` too? Or will the generator fail altogether right now?

Reply:
Serialization won't work because the serializer is looked up by runtimeType. You could add your own serializer at runtime, I think.

The value types will work--but will try hard to return to using the standard `BuiltSet`, because whenever a builder is instantiated for you it's the standard builder.

There's no need to implement `Built`, the current built_collections don't. (And can't, it would create a dependency cycle).